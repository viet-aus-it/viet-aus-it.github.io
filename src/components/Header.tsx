import { FC } from 'react';
import { Link } from 'gatsby';
import { rhythm, scale } from '../utils/typography';

interface HeaderProps {
  isRootPath: boolean;
  title: string;
}

const Header: FC<HeaderProps> = ({ isRootPath, title }) => {
  const LinkContent = (
    <Link
      style={{
        boxShadow: `none`,
        color: `inherit`,
      }}
      to="/"
    >
      {title}
    </Link>
  );

  const RootHeader = (
    <h1
      style={{
        ...scale(1.5),
        marginBottom: rhythm(1.5),
        marginTop: 0,
      }}
    >
      {LinkContent}
    </h1>
  );

  const PageHeader = (
    <h3
      style={{
        fontFamily: `Montserrat, sans-serif`,
        marginTop: 0,
      }}
    >
      {LinkContent}
    </h3>
  );

  const HeaderContent = isRootPath ? RootHeader : PageHeader;

  return <header>{HeaderContent}</header>;
};

export default Header;
