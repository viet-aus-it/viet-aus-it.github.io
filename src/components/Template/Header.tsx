import { Link } from 'gatsby';
import { rhythm, scale } from '../../utils/typography';

interface HeaderProps {
  isRootPath: boolean;
  title: string;
}

function Header({ isRootPath, title }: HeaderProps) {
  const LinkContent = (
    <Link
      style={{
        boxShadow: `none`,
        color: `inherit`,
      }}
      to="/"
    >
      {title}
    </Link>
  );

  const HeaderContent = isRootPath ? (
    <h1
      style={{
        ...scale(1.5),
        marginBottom: rhythm(1.5),
        marginTop: 0,
      }}
    >
      {LinkContent}
    </h1>
  ) : (
    <h3
      style={{
        fontFamily: `Montserrat, sans-serif`,
        marginTop: 0,
      }}
    >
      {LinkContent}
    </h3>
  );

  return <header>{HeaderContent}</header>;
}

export default Header;
