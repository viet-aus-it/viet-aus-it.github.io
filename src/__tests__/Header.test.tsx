/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Template/Header';

describe('Header', () => {
  it('should render without crashing', () => {
    const treeRootPath = renderer
      .create(<Header isRootPath title="Viet-Aus-IT Blog" />)
      .toJSON();
    expect(treeRootPath).toMatchSnapshot();

    const treeNotRoot = renderer
      .create(<Header isRootPath={false} title="Viet-Aus-IT Blog" />)
      .toJSON();
    expect(treeNotRoot).toMatchSnapshot();
  });
});
