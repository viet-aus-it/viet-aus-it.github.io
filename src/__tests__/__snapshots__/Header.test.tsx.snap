// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without crashing 1`] = `
<header>
  <h1
    style={
      Object {
        "fontSize": "3.95285rem",
        "lineHeight": "4.375rem",
        "marginBottom": "2.625rem",
        "marginTop": 0,
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "boxShadow": "none",
          "color": "inherit",
        }
      }
    >
      Viet-Aus-IT Blog
    </a>
  </h1>
</header>
`;

exports[`Header should render without crashing 2`] = `
<header>
  <h3
    style={
      Object {
        "fontFamily": "Montserrat, sans-serif",
        "marginTop": 0,
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "boxShadow": "none",
          "color": "inherit",
        }
      }
    >
      Viet-Aus-IT Blog
    </a>
  </h3>
</header>
`;
